### DISCUSSION TITLE
Jotform Integration with Supabase

### DISCUSSION THREAD

**Discussion by Tria80** [supabase/supabase#29459] [Questions]
Hi Lovely Community! 
Hope someone can help me out of my misery! :) 

Very newby to supabase! So please bear with me.
Tried an Edge function to insert data from **a Jotform submission into a Supabase table.** Is this the best manner for this integration? ( will the API route be better?) Either way, struggling with both.
The webhook on the Jotform is working fine, but not getting data into the table! I've spend hours with Claude today, think the problem might be in the security somewhere? Seems to be such a simple integration!! Help!

**My .ts for Edge function:**
import { serve } from "https://deno.land/std@0.131.0/http/server.ts";
import { createClient } from "https://esm.sh/@supabase/supabase-js@2";

// Initialize the Supabase client
const supabaseUrl = Deno.env.get('https://tuhdqktxkiwhbcvhkfyu.supabase.co');
const supabaseKey = Deno.env.get('eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InR1aGRxa3R4a2l3aGJjdmhrZnl1Iiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTcxNTQxOTMwNiwiZXhwIjoyMDMwOTk1MzA2fQ.NHjU-KrWqlIQJ6-4wf0r83PZ0wmr234vHE8w0jc_81g');
const supabase = createClient(supabaseUrl, supabaseKey);
const secretKey = Deno.env.get('iGE7YBldeUr/NJ/HO/G8Vgf5BMBrneQ0HfO1wn1Vd/n50GrBfOybCy84odXRwyKgFBSbL2ut/vl3yXwpDkFK/Q==');

serve(async (req) => {
  if (req.method !== "POST") {
    return new Response(JSON.stringify({ error: "Method not allowed" }), { status: 405 });
  }

  try {
    // Parse the JSON payload from Jotform
    const formData = await req.json();

    // Insert the form data into the Supabase table
    const { data, error } = await supabase
      .from('patient')
      .insert([
        {
          patient_name: formData.name,        // Assuming Jotform sends 'name'
          patient_email: formData.email,      // Assuming Jotform sends 'email'
          patient_surname: formData.surname,  // Assuming Jotform sends 'surname'
        }
      ]);

    if (error) {
      return new Response(JSON.stringify({ error: 'Failed to insert data', details: error.message }), { status: 500 });
    }

    return new Response(JSON.stringify({ message: 'Data saved successfully', data }), { status: 200 });
  } catch (err) {
    return new Response(JSON.stringify({ error: 'Invalid request', details: err.message }), { status: 400 });
  }
});

**Comment by kimnguan** [comment]
I don't get anything into Supabase neither.
